variables.tf

```
@@ -144,6 +144,27 @@ variable "managed_rules" {
   default = {}
 }
 
+variable "recording_mode" {
+  description = <<-DOC
+    The mode for AWS Config to record configuration changes. Possible values are:
+    - Continuous
+    - Daily
+
+    You can also override the recording frequency for specific resource types.
+    See the following for more information:
+    https://docs.aws.amazon.com/config/latest/developerguide/stop-start-recorder.html
+  DOC
+  type = object({
+    recording_frequency     = string
+    recording_mode_override = optional(list(object({
+      description         = string
+      recording_frequency = string
+      resource_types      = list(string)
+    })))
```

AFAICT, it seems there can only be one `recording_mode_override` block, yeah? Considering that, I'd just make this a single option object like so: 

```suggestion
    recording_mode_override = optional(object({
      description         = string
      recording_frequency = string
      resource_types      = list(string)
    }))
```

This way we don't need to worry about list access and can avoid that data type. Or am I missing something on that?

---

The [AWS API has a list](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configservice/put-configuration-recorder.html#:~:text=to%20Continuous%20recording.-,recordingModeOverrides,-%2D%3E%20(list)), which means it is meant for future expansion. Whomever designed the [Terraform resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/config_configuration_recorder#recording_mode-configuration-block) did not realize that and just coded a structure.

We can align to the Terraform resource, knowing that in the future this could be an issue.